blocks/blockstest/blocks.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// Package blockstest provides test helpers for constructing [Streaming
+// Asynchronous Execution] (SAE) blocks.
+//
+// [Streaming Asynchronous Execution]: https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/194-streaming-asynchronous-execution
+package blockstest
+
+import (
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/libevm/core"
+	"github.com/ava-labs/libevm/core/state"
+	"github.com/ava-labs/libevm/core/types"
+	"github.com/ava-labs/libevm/ethdb"
+	"github.com/ava-labs/libevm/libevm/options"
+	"github.com/ava-labs/libevm/params"
+	"github.com/ava-labs/libevm/triedb"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/strevm/blocks"
+	"github.com/ava-labs/strevm/gastime"
+	"github.com/ava-labs/strevm/saetest"
+)
+
+// An EthBlockOption configures the [types.Header] created by [NewEthBlock]. It
+// SHOULD NOT modify the `Number` and `ParentHash`, but MAY modify any other
+// field.
+type EthBlockOption = options.Option[types.Header]
+
+// NewEthBlock constructs a raw Ethereum block with the given arguments.
+func NewEthBlock(parent *types.Block, txs types.Transactions, opts ...EthBlockOption) *types.Block {
+	hdr := &types.Header{
+		Number:     new(big.Int).Add(parent.Number(), big.NewInt(1)),
+		ParentHash: parent.Hash(),
+		BaseFee:    big.NewInt(0),
+	}
+	hdr = options.ApplyTo(hdr, opts...)
+	return types.NewBlock(hdr, txs, nil, nil, saetest.TrieHasher())
+}
+
+// ModifyHeader returns an option to modify the [types.Header] constructed by
+// [NewEthBlock]. See [EthBlockOption] for caveats.
+func ModifyHeader(fn func(*types.Header)) EthBlockOption {
+	return options.Func[types.Header](fn)
+}
+
+// NewBlock constructs an SAE block, wrapping the raw Ethereum block.
+func NewBlock(tb testing.TB, eth *types.Block, parent, lastSettled *blocks.Block) *blocks.Block {
+	tb.Helper()
+	b, err := blocks.New(eth, parent, lastSettled, saetest.NewTBLogger(tb, logging.Warn))
+	require.NoError(tb, err, "New()")
```

Comment:
```suggestion
	require.NoError(tb, err, "blocks.New()")
```
`New` by itself doesn't seem informative